src/storage/events/jwks-create-signing-secret.ts

```diff
@@ -0,0 +1,60 @@
+import { BaseEvent } from './base-event'
+import { Job, SendOptions, WorkOptions } from 'pg-boss'
+import { logger, logSchema } from '@internal/monitoring'
+import { BasePayload } from '@internal/queue'
+import { generateUrlSigningJwk } from '@internal/database'
+
+interface JwksCreateSigningSecretPayload extends BasePayload {
+  tenantId: string
+}
+
+export class JwksCreateSigningSecret extends BaseEvent<JwksCreateSigningSecretPayload> {
+  static queueName = 'tenants-jwks-create'
+
+  static getWorkerOptions(): WorkOptions {
+    return {
+      teamSize: 200,
+      teamConcurrency: 10,
+      includeMetadata: true,
+    }
+  }
+
+  static getQueueOptions(payload: JwksCreateSigningSecretPayload): SendOptions {
+    return {
+      expireInHours: 2,
+      singletonKey: payload.tenantId,
+      useSingletonQueue: true,
+      retryLimit: 3,
+      retryDelay: 5,
+      priority: 10,
+    }
+  }
+
+  static async handle(job: Job<JwksCreateSigningSecretPayload>) {
+    const { tenantId } = job.data
+
+    try {
+      const { kid } = await generateUrlSigningJwk(tenantId)
```

Comment:
this method doesn't seem to be idempotent, what happens if this job is called twice?

Reply:
Good call. Updated.